scripts/visualize/visualize_utils.R

```diff
@@ -151,25 +156,31 @@ clip_sp.SpatialLines <- function(sp, xlim, ylim, ...){
 clip_sp.Spatial <- function(sp, xlim, ylim, ..., clip.fun = rgeos::gIntersection){
   
   clip <- as.sp_box(xlim, ylim, sp::CRS(sp::proj4string(sp)))
-  g.i <- rgeos::gIntersects(sp, clip, byid = T) 
-  
-  sp <- rgeos::gBuffer(sp, byid=TRUE, width=0) # zero buffer to avoid TopologyException
-  has.data <- ("data" %in% slotNames(sp))
-
-  out <- lapply(which(g.i), function(i) {
-    g.out <- clip.fun(sp[i,], clip)
-    row.names(g.out) <- row.names(sp)[i]
-    return(g.out)
-  })
   
-  # use rbind.SpatialPolygons method to combine into a new object.
-  clipped.sp <- do.call("rbind", out)
   
-  if(has.data){
-    data.out <- as.data.frame(sp)[g.i, ]
-    row.names(data.out) <- row.names(sp)[g.i]
-    clipped.sp <- as(object = clipped.sp, Class = class(sp))
-    clipped.sp@data <- data.out
+  if (!inherits(sp, 'SpatialPoints')){
+    g.i <- rgeos::gIntersects(sp, clip, byid = T) 
```

Comment:
The sp::over function does the same as gIntersects and has some nice handling for data frames. Not for now, but note that using that in the future could avoid some headaches... I think.

Reply:
There was some reason I was swayed away from that by Google...not sure. But as noted above, this needs refactor when there is time